src/village/mod.rs

```diff
@@ -79,43 +75,62 @@ mod tests {
     use village::*;
 
     #[test]
-    fn disallow_adding_existing_resource() {
+    fn add_resources() {
         let mut v = Village::new(1);
-        v.add_resource(ResourceType::Gold);
-        v.add_resource(ResourceType::Gold);
+        v.create_resource(ResourceType::Gold);
+        v.create_resource(ResourceType::Gold);
+        v.create_resource(ResourceType::Wood);
 
-        assert_eq!(1, v.resource_types().len());
+        assert_eq!(3, v.all_resources().len());
     }
 
     #[test]
     fn assign_workers_to_resource() {
         let mut v = Village::new(4);
-        v.add_resource(ResourceType::Wood);
-        v.assign_workers(ResourceType::Wood, 1);
+        v.create_resource(ResourceType::Gold);
+        v.create_resource(ResourceType::Gold);
+        v.create_resource(ResourceType::Wood);
+        let mut resource_assignments: Vec<(u32, u32)> = Vec::new();
```

Comment:
resource_assignments makes this more complicated than we need. Just use hard coded values in the assign_workers calls.